### ISSUE TITLE
一个bug

### ISSUE THREAD

**Issue by pSGAme** [AuroraZengfh/Local-Prompt#6]
在localprompt.py里面，test的函数测试的时候，设置了self.set_model_mode("eval"), 然而，这只是将prompt_learner的状态设置为eval，整个模型的状态还是training的状态。因此，需要显式加一句self.model.eval().

此外，我在跑这份代码的时候碰到一件很奇怪的事情。在训练的时候，测试的最后的一个epoch的 ID accuracy 和 直接eval_ood_detection.py里跑出来的ID accuracy 结果不太一样。想问问是为什么呀？

**Comment by AuroraZengfh** [comment]
Thanks for you interest and attention of our paper.

For the former, it may depend on the version of used libs; for the later, we simply use the default function of ID accuracy calculation in Dassl during training and use the proposed way of ID accuracy calculation in inference, so the ID accuracy should have slight difference. You can view the location where the forward function runs by setting breakpoints.

**Comment by pSGAme** [comment]
Thank you for the detailed explanation. I've identified the root cause: in the released scripts, top-k is set to 50 during training (train.sh), but it's set to 10 during evaluation (eval.sh). This discrepancy explains the issue. I will go ahead and close this now.
